packages/svelte/src/compiler/phases/3-transform/server/visitors/shared/component.js

```diff
@@ -318,16 +319,21 @@ export function build_inline_component(node, expression, context) {
 	}
 
 	if (optimiser.expressions.length > 0) {
-		statement = call_child_renderer(b.block([optimiser.apply(), statement]), true);
+		statement = create_async_block(b.block([optimiser.apply(), statement]));
 	}
 
+	// TODO in the async case this probably needs to go inside the async block?
```

Comment:
TODO

Reply:
yeah looked into this, nothing to do here